src/units/unit.hpp

```diff
@@ -1737,11 +1737,11 @@ class unit : public std::enable_shared_from_this<unit>
 
 	/** Checks whether this unit currently possesses a given ability used like weapon
 	 * @return True if the ability @a tag_name is active.
-	 * @param special the const config to one of abilities @a tag_name checked.
+	 * @param ability the const config to one of abilities @a tag_name checked.
 	 * @param tag_name name of ability type checked.
 	 * @param loc location of the unit checked.
 	 */
-	bool get_self_ability_bool(const config& special, const std::string& tag_name, const map_location& loc) const;
+	bool get_self_ability_bool(const config& ability, const std::string& tag_name, const map_location& loc) const {return get_self_ability_bool_impl(ability, tag_name, loc);}
```

Comment:
If `get_self_ability_bool_impl` takes exactly the same arguments as `get_self_ability_bool`, and returns the same status, why are they two separate functions?

Reply:
i want what get_abilities() and get_ability_bool() use private functions, but get_self_ability_bool must be public for ability weapon and filters.cpp

Reply:
Why? What's gained by having separate functions?

Reply:
do you mind that get_abilities checking uses a public function? If so I will only use one function.

Reply:
I see no reason not to use the public function.

My answer would be different if the public functions were virtual, but they aren't. It would also change if those functions needed recursion guards themselves, but I think those are only needed for checking whether an ability is active.

Reply:
private function removed